shared/chat/conversation/list/index.desktop.js

```diff
@@ -1,244 +1,269 @@
-    }
+  _rowRenderer = ({index, isScrolling, isVisible, key, parent, style}) => {
+    const messageKey = this.props.messageKeys.get(index)
+    const prevMessageKey = this.props.messageKeys.get(index - 1)
+    const isSelected = messageKey === this.state.selectedMessageKey
+    return (
+      <Virtualized.CellMeasurer
+        cache={this._cellCache}
+        columnIndex={0}
+        key={key}
+        parent={parent}
+        rowIndex={index}>
+        {({measure}) => {
+          const message = messageFactory(messageKey, prevMessageKey, this._onAction, this._onShowEditor, isSelected, measure)
+          return (
+            <div style={style}>
+              {message}
+            </div>
+          )
+        }}
+      </Virtualized.CellMeasurer>
+    )
+  }
 
-    if (this.props.moreToLoad !== nextProps.moreToLoad) {
-      this._shouldForceUpdateGrid = true
+  _onCopyCapture (e) {
+    // Copy text only, not HTML/styling.
+    e.preventDefault()
+    clipboard.writeText(window.getSelection().toString())
+  }
+
+  _handleListClick = () => {
+    if (window.getSelection().isCollapsed) {
+      this.props.onFocusInput()
     }
   }
 
-  _invalidateChangedMessages (props: Props) {
-    const headerMessagesCount = props.headerMessages.count()
-    this.state.messages.forEach((item, index) => {
-      const oldMessage = props.messages.get(index, {})
-
-      if (item.type === 'Text' && oldMessage.type === 'Text' &&
-        (item.messageState !== oldMessage.messageState ||
-        item.editedCount !== oldMessage.editedCount)
-      ) {
-        this._toRemeasure.push(index + headerMessagesCount)
-      } else if (item.type === 'Attachment' && oldMessage.type === 'Attachment' &&
-                 (item.previewPath !== oldMessage.previewPath ||
-                  !shallowEqual(item.previewSize, oldMessage.previewSize))) {
-        this._toRemeasure.push(index + headerMessagesCount)
-      } else if (!shallowEqual(item, oldMessage)) {
-        this._shouldForceUpdateGrid = true
-      }
-    })
+  _onRowsRendered = ({stopIndex}: {stopIndex: number}) => {
+    this._lastRowIdx = stopIndex
+  }
+
+  _setListRef = (r: any) => {
+    this._list = r
   }
 
-  _updateInternalMessages = (props: Props) => {
-    if (props.messages !== this.state.messages) {
-      this._invalidateChangedMessages(props)
-      this.setState({
-        messages: props.messages,
-      })
+  render () {
+    if (!this.props.validated) {
+      return (
+        <div style={{alignItems: 'center', display: 'flex', flex: 1, justifyContent: 'center'}}>
+          <Icon type='icon-securing-266' style={{alignSelf: 'flex-start'}} />
+        </div>
+      )
     }
+
+    const rowCount = this.props.messageKeys.count()
+    const scrollToIndex = this.state.isLockedToBottom ? rowCount - 1 : this._keepIdxVisible
+
+    // We pass additional props (listRerender, selectedMessageKey) to Virtualized.List so we can force re-rendering automatically
```

Comment:
this way we don't need to ever call forceGridUpdate or any of that from the old code